### PR TITLE
Adding support for alter with soup spec in hybrid and react native apps

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

**Comment by bhariharan** [comment]
👍

File: libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SmartStorePlugin.java [hunk 1]
```diff
@@ -59,25 +59,11 @@
 public class SmartStorePlugin extends ForcePlugin {
 
 	// Keys in json from/to javascript
-	public static final String BEGIN_KEY = "beginKey";
```
Comment (huminzhi) [inline libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SmartStorePlugin.java]: Just curious @wmathurin if it's ok to remove all those keys in terms of upgrading..the change should not impact upgrading right?

Comment (wmathurin) [inline libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SmartStorePlugin.java]: These constants moved to QuerySpec.java (
https://forcedotcom.github.io/SalesforceMobileSDK-Android/com/salesforce/androidsdk/smartstore/store/QuerySpec.html)
 many releases back.

**Comment by bhariharan** [comment]
👍

File: libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SmartStorePlugin.java (hunk 1 above)
Comment (huminzhi) [inline libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SmartStorePlugin.java]: Thank you!

**Comment by wmathurin** [comment]
Actually this PR was not finished.
I'll send another one soon with the remaining work.

**Pull request merged**